staging/src/k8s.io/kubectl/pkg/cmd/apply/apply.go

```diff
@@ -618,16 +619,16 @@ See https://kubernetes.io/docs/reference/using-api/server-side-apply/#conflicts`
 		if err != nil {
 			return err
 		}
-		patchBytes, patchedObject, err := patcher.Patch(info.Object, modified, info.Source, info.Namespace, info.Name, o.ErrOut)
+		patchedObject, err := patcher.Patch(info.Object, modified, info.Source, info.Namespace, info.Name, o.ErrOut)
 		if err != nil {
-			return cmdutil.AddSourceToErr(fmt.Sprintf("applying patch:\n%s\nto:\n%v\nfor:", patchBytes, info), info.Source, err)
+			return cmdutil.AddSourceToErr(fmt.Sprintf("applying patch:\n%v\nto:\n%v\nfor:", patchedObject, info), info.Source, err)
 		}
 
 		info.Refresh(patchedObject, true)
 
 		WarnIfDeleting(info.Object, o.ErrOut)
 
-		if string(patchBytes) == "{}" && !o.shouldPrintObject() {
+		if reflect.DeepEqual(patchedObject, info.Object) && !o.shouldPrintObject() {
```

Comment:
In the case the `DeepEqual` evaluates to false, this comparison could be far slower. 

For this use case within kubectl, this is likely OK. But it is worth pointing out.